projects/vdk-core/.gitlab-ci.yml

```diff
@@ -83,12 +83,60 @@ vdk-core-simple_func_test_with_py310:
   image: "python:3.10"
   extends: .vdk-core-simple_func_test
 
+
+vdk-core-release-candidate:
+  stage: pre_release
+  before_script:
+    - cd projects/vdk-core/
+  script:
+    - pip install -U pip setuptools wheel twine
+    - export VDK_RELEASE_OVERRIDE_PATCH_VERSION=${CI_PIPELINE_ID}.dev${CI_PIPELINE_IID}
+    - ./cicd/set-patch-version.sh "$VDK_RELEASE_OVERRIDE_PATCH_VERSION"
+    - ./cicd/release-vdk-core.sh
+  retry: !reference [.retry, retry_options]
+  only:
+    refs:
+      - main
+    changes: *vdk_core_locations
+
+
+vdk-core-release-candidate-base-image:
+  stage: pre_release_image
+  image: docker:19.03.8
+  services:
+    - docker:19.03.8-dind
+  variables:
+    DOCKER_HOST: tcp://localhost:2375
+    DOCKER_DRIVER: overlay2
+    DOCKER_TLS_CERTDIR: ""
+  before_script:
+    - cd projects/vdk-core/
+  script:
+    - echo "Build and release candidate vdk image of vdk-core"
+    - apk add --no-cache git bash python3
+    - python3 -V
+    - python3 -m pip install -U setuptools configparser
+    - export VDK_RELEASE_OVERRIDE_PATCH_VERSION=${CI_PIPELINE_ID}.dev${CI_PIPELINE_IID}
+    - ./cicd/set-patch-version.sh "$VDK_RELEASE_OVERRIDE_PATCH_VERSION"
+    - export VDK_PACKAGE=$(python3 setup.py --name)
+    - export VDK_VERSION=$(python3 setup.py --version)
+    - export BUILD_TYPE=pre_release
+    - bash -x ./cicd/deploy-base-vdk-image.sh
```

Comment:
This is base image with only vdk-core (without any plugins). That means for vdk-heartbeat I think we can only use "empty" type of test (data job that does nothing - just prints)

 https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-heartbeat/src/vdk/internal/heartbeat/config.py#L167  
 
 
 Which I think it's good enough actually as it would test vdk-core efficiently.